stencil/src/components/pulumi-api-doc-nav-node/pulumi-api-doc-nav-node.tsx

```diff
@@ -0,0 +1,135 @@
+import { Component, h, Listen, Prop } from "@stencil/core";
+import { APINavNode, APINavNodeType } from "../pulumi-api-doc-filterable-nav/pulumi-api-doc-filterable-nav";
+
+@Component({
+    tag: "pulumi-api-doc-nav-node",
+    styleUrl: "pulumi-api-doc-nav-node.scss",
+    shadow: false,
+})
+export class PulumiApiDocNavNode {
+    @Prop()
+    node: APINavNode;
+
+    @Prop()
+    href: string;
+
+    @Prop()
+    depth: number = 0;
+
+    @Prop({ mutable: true })
+    isExpanded: boolean;
+
+    @Listen("expanded-change", { target: "document" })
+    onExpansionChange(event: CustomEvent) {
+        if (this.node.name === event.detail.id) {
+            this.isExpanded = event.detail.expanded;
+        }
+    }
+
+    componentWillLoad() {
+        this.isExpanded = !!this.node.isExpanded || this.isNodeInPathForCurrentlyVisiblePage(this.node.name);
+    }
+
+    componentShouldUpdate(newVal, oldVal, propName) {
+        if (propName === "isExpanded"){
```

Comment:
This confuses me a bit, wouldn't we always want to return something here? Could we satisfy that and also simplify this to:
```
// I'm not sure what this comparison represents conceptually so this could help the reader IMO
const meaningfulBoolName = odlValue === true && newValue === false;
return propName === "isExpanded" && !meaningfulBoolName;
```

Reply:
I believe this is so that we don't throw out rendered nodes on collapse. I added a comment that clarifies this:

```
        // When the node is expanded, we render child nodes.
        // When we collapse that same node, we want to avoid throwing out
        // the rendered nodes. (Helps with toggling nodes open/close repeatedly)
```

Will go ahead and merge with this, but we can follow up if @kimberleyamackenzie clarifies that the intent is something else..